ui/site/src/util.js

```diff
@@ -5,23 +5,30 @@ lichess.engineName = 'Stockfish 8';
 lichess.raf = (window.requestAnimationFrame || window.setTimeout).bind(window);
 lichess.requestIdleCallback = (window.requestIdleCallback || window.setTimeout).bind(window);
 lichess.storage = (function() {
-  var withStorage = function(f) {
-    // can throw an exception when storage is full
-    try {
-      return !!window.localStorage ? f(window.localStorage) : null;
-    } catch (e) {}
-  }
+  try {
+    // just accessing localStorage can throw an exception...
+    var storage = window.localStorage;
+  } catch (e) {}
+  var withStorage = storage ? function(f) {
+    // this can throw exceptions as well.
+    try { return f(storage); }
+    catch (e) {}
+  } : function() {};
```

Comment:
you said in slack that grabbing localStorage each time was unnecessary work... And it is significantly slower than local variable access due to a security check.

We have to wrap ops in try because quota or other exceptions can occur at any time.